app/assets/javascripts/models/ui/collections_list.js.coffee

```diff
@@ -67,11 +67,20 @@ ns.CollectionsList = do ($ = jQuery
       query = @query.params()
       collections = @collections.results().map (collection) -> collection.id
       index = @collections.results().indexOf(collection)
+
+      exact_match = switch query.free_text
```

Comment:
The strings should be compared in lower (or upper) cases. So AE_Land matches ae_land.